celery/concurrency/asynpool.py

```diff
@@ -89,8 +90,7 @@ def unpack_from(fmt, iobuf, unpack=unpack):  # noqa
 
 def gen_not_started(gen):
     """Return true if generator is not started."""
-    # gi_frame is None when generator stopped.
-    return gen.gi_frame and gen.gi_frame.f_lasti == -1
+    return inspect.getgeneratorstate(gen) == "GEN_CREATED"
```

Comment:
is this change backward compatible?

Reply:
Yes - `inspect.getgeneratorstate` is available since Python 3.2 and this function returns `True` in the same cases as the previous implementation (when the generator is created, but not yet started).

There is some testing of this in `test_AsynPool::test_gen_not_started`. I also did a bit more [manual testing](https://gist.github.com/c13541de1cd0426111ab8bcf58b0aaa5) with Python 3.7, 3.10, and PyPy 7.3.9.

Reply:
is it possible to add some additonal testing for the change?

Reply:
Sure. I added assertions for a few additional cases:

- while the generator is running
- after the generator exited due to an exception

That's all the cases I could think of.